compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -82,6 +83,13 @@ open class ClassCodegen protected constructor(
             else -> null
         }
 
+    fun getRegeneratedObjectNameGenerator(function: IrFunction): NameGenerator {
+        val name = function.name.takeIf { !it.isSpecial }
```

Comment:
This should fix tests on Windows.

Reply:
maybe smth like val name = function.name.takeIf { !it.isSpecial } ?: "_special_"

Reply:
done